asv_bench/benchmarks/timeseries.py

```diff
@@ -10,1116 +10,312 @@
 
-    def time_timeseries_iter_periodindex_preexit(self):
-        self.iter_n(self.idx2, self.M)
-
     def iter_n(self, iterable, n=None):
         self.i = 0
         for _ in iterable:
             self.i += 1
             if ((n is not None) and (self.i > n)):
                 break
 
+    def time_iter_datetimeindex(self):
+        self.iter_n(self.idx1)
 
-class timeseries_large_lookup_value(object):
-    goal_time = 0.2
+    def time_iter_datetimeindex_preexit(self):
+        self.iter_n(self.idx1, self.M)
 
-    def setup(self):
-        self.N = 100000
-        self.rng = date_range(start='1/1/2000', periods=self.N, freq='T')
-        if hasattr(Series, 'convert'):
-            Series.resample = Series.convert
-        self.ts = Series(np.random.randn(self.N), index=self.rng)
-        self.rng = date_range(start='1/1/2000', periods=1500000, freq='S')
-        self.ts = Series(1, index=self.rng)
+    def time_iter_periodindex(self):
+        self.iter_n(self.idx2)
+
+    def time_iter_periodindex_preexit(self):
+        self.iter_n(self.idx2, self.M)
 
-    def time_timeseries_large_lookup_value(self):
-        self.ts[self.ts.index[(len(self.ts) // 2)]]
-        self.ts.index._cleanup()
 
+#----------------------------------------------------------------------
+# Resampling
 
-class timeseries_period_downsample_mean(object):
+class ResampleDataFrame(object):
     goal_time = 0.2
 
     def setup(self):
-        self.N = 100000
-        self.rng = date_range(start='1/1/2000', periods=self.N, freq='T')
-        if hasattr(Series, 'convert'):
-            Series.resample = Series.convert
-        self.ts = Series(np.random.randn(self.N), index=self.rng)
-        self.rng = period_range(start='1/1/2000', end='1/1/2001', freq='T')
-        self.ts = Series(np.random.randn(len(self.rng)), index=self.rng)
+        self.rng = date_range(start='20130101', periods=100000, freq='50L')
+        self.df = DataFrame(np.random.randn(100000, 2), index=self.rng)
 
-    def time_timeseries_period_downsample_mean(self):
-        self.ts.resample('D', how='mean')
+    def time_max_numpy(self):
+        self.df.resample('1s', how=np.max)
 
+    def time_max_string(self):
+        self.df.resample('1s', how='max')
 
-class timeseries_resample_datetime64(object):
-    goal_time = 0.2
+    def time_mean_numpy(self):
+        self.df.resample('1s', how=np.mean)
 
-    def setup(self):
-        self.N = 100000
-        self.rng = date_range(start='1/1/2000', periods=self.N, freq='T')
-        if hasattr(Series, 'convert'):
-            Series.resample = Series.convert
-        self.ts = Series(np.random.randn(self.N), index=self.rng)
-        self.rng = date_range(start='2000-01-01 00:00:00', end='2000-01-01 10:00:00', freq='555000U')
-        self.int_ts = Series(5, self.rng, dtype='int64')
-        self.ts = self.int_ts.astype('datetime64[ns]')
+    def time_mean_string(self):
+        self.df.resample('1s', how='mean')
```

Comment:
yes, I am not sure yet how to approach this. Because we also want that the tests (generally) keep running on older versions to see the evolution over time (at least for some time). So I would personally not yet change it now.

So either we change the benchmarks only when we actually remove the old syntax, or we could define some wrapper method specifically here in the bench file that calls the correct thing

```
def resample_mean(df, ...):
    if pd.__version__ < ..:
        df.resample('', how='mean')
    else:
        df.resample().mean()
```

But that is also a bit unwieldy

We could of course add benchmarks with the new syntax for things that were not possible before with the old syntax.